src/reducers/LoginReducer.jsx

```diff
@@ -31,6 +31,17 @@ const LoginReducer = (state = initialState, action) => {
         allCourses: action.courses,
         loggedIn: true,
       });
+    case 'CHANGE_SEMESTER': {
+      const isCourseMatch = course => course.courseId === action.courseId;
+      const index = state.plannedCourses.findIndex(isCourseMatch);
+      const course = state.plannedCourses.find(isCourseMatch);
+      course.semester = action.newSemester;
+      const updatedPlanned = state.plannedCourses;
+      updatedPlanned[index] = course;
+      return Object.assign({}, state, {
+        plannedCourses: updatedPlanned,
+      });
+    }
```

Comment:
I'm confused, isn't the Redux state global?  So you should be able to access `state.plannedCourses` from any reducer, and then separating out this `CHANGE_SEMESTER` reducer case should be easy.